internal/controller/profiles/monitoring/monitoring.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Apache Software Foundation (ASF)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+
+	operatorapi "github.com/apache/incubator-kie-kogito-serverless-operator/api/v1alpha08"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/internal/controller/monitoring"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/internal/controller/profiles/common"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/log"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ MonitoringEventingHandler = &monitoringObjectManager{}
+
+type monitoringObjectManager struct {
+	serviceMonitor common.ObjectEnsurer
+	*common.StateSupport
+}
+
+func NewMonitoringHandler(support *common.StateSupport) MonitoringEventingHandler {
+	return &monitoringObjectManager{
+		serviceMonitor: common.NewObjectEnsurer(support.C, common.ServiceMonitorCreator),
+		StateSupport:   support,
+	}
+}
+
+type MonitoringEventingHandler interface {
+	Ensure(ctx context.Context, workflow *operatorapi.SonataFlow) ([]client.Object, error)
+}
+
+func (k monitoringObjectManager) Ensure(ctx context.Context, workflow *operatorapi.SonataFlow) ([]client.Object, error) {
+	var objs []client.Object
+	monitoringAvail, err := monitoring.GetPrometheusAvailability(k.Cfg)
+	if err != nil {
+		klog.V(log.I).InfoS("Error checking Prometheus availability: %v", err)
+		return nil, err
+	}
+	if !monitoringAvail {
+		klog.V(log.I).InfoS("Prometheus is not installed")
```

Comment:
In the platform controller, if the `.spec.monitoring.enabled` is true and Prometheus is not available, we should emit a warning event. So we can bring this check there.

Here, you can skip logging it.

Reply:
Good idea. Done.